libs/partners/anthropic/langchain_anthropic/chat_models.py

```diff
@@ -692,14 +699,20 @@ async def _astream(
                 message_chunk = AIMessageChunk(
                     content=message.content,
                     tool_call_chunks=tool_call_chunks,  # type: ignore[arg-type]
+                    usage_metadata=message.usage_metadata,
                 )
                 yield ChatGenerationChunk(message=message_chunk)
             else:
                 yield cast(ChatGenerationChunk, result.generations[0])
             return
+        full_generation_info: dict = {}
         async with self._async_client.messages.stream(**params) as stream:
             async for text in stream.text_stream:
```

Comment:
Just flagging - for streaming tool calls, we'll need to handle the raw events instead of the text stream. Worth switching it out now?